### DISCUSSION TITLE
Fields Calculator Guide

### DISCUSSION THREAD

**Discussion by bakerwillis** [ansys/pyaedt#4388] [Q&A] [answered]
Hi Everyone, 

I have been using this pyaedt tool, and it works great. And Have found the documentation mostly helpful and can find what I need to get something done. This is true except for the fields calculator. The only tips i can find on how to use this is just from the snippets in the online documentaion. Is there a place where i can find how to access all the various fields within the fields calculator? 

for instance, i would like to do something like this in pyaedt:

create an expression which is the dot product of some field, say the H-field, along a unit vector of a specified coordinate system

![Capture](https://github.com/ansys/pyaedt/assets/8380510/2a8c6ea5-5fc1-4287-b4aa-c9ef1448fc5a)

then plot this value along a specified path which has been loaded previous. 

Thanks!

-Will

**Comment by gmalinve** [comment, marked as answer]
Hi @bakerwillis, 

Thanks for submitting your question.
There's no documentation because we don't cover all existing functionalities of the field calculator with PyAEDT.
What we can suggest you to do is to record the scripts while doing fields calculator operation and the re-use the recorded commands with PyAEDT by accessng the `ofieldsreporter`.
For example:
`app.ofieldsreporter.CalcOp("Value")`
`app.post.ofieldsreporter.CalcOp("Value")`
Please also refer to this example:
[Maxwell 3D: magnet DC analysis](https://aedt.docs.pyansys.com/version/stable/examples/03-Maxwell/Maxwell_Magnet.html#)

Hope this helps,
Please do let me know if I answered your question and close this discussion!

Thanks,

Giulia

**Reply by bakerwillis** [reply]
ok, thanks for the reply, I will try this.

**Reply by bakerwillis** [reply]
This worked perfect! Thanks again.
